components/hab/src/cli/hab/util.rs

```diff
@@ -18,29 +27,99 @@ use structopt::StructOpt;
+    #[structopt(name = "BLDR_URL",
+                short = "u",
+                long = "url",
+                validator = valid_url)]
+    pub value: Option<Url>,
+}
+
+#[derive(ConfigOpt, StructOpt, Deserialize, Serialize)]
+#[structopt(no_version)]
+#[configopt(derive(Serialize))]
+#[allow(dead_code)]
+pub struct BldrOrigin {
+    /// The Builder origin name to target
+    #[structopt(name = "ORIGIN", short = "o", long = "origin", validator = valid_origin)]
+    pub name: Origin,
+}
+
+pub fn bldr_url_from_env_load_or_default() -> String {
+    bldr_url_from_env().unwrap_or_else(|| {
+                           match config::load() {
+                               Ok(config) => {
+                                   match config.bldr_url {
+                                       Some(v) => v,
+                                       None => DEFAULT_BLDR_URL.to_string(),
+                                   }
+                               }
+                               Err(e) => {
+                                   println!("Found a cli.toml but unable to load it. Resorting \
+                                             to default BLDR_URL: {}",
+                                            e);
+                                   DEFAULT_BLDR_URL.to_string()
+                               }
+                           }
+                       })
+}
+
+pub fn bldr_url_from_args_env_load_or_default(opt: Option<Url>) -> String {
+    if let Some(url) = opt {
+        url.to_string()
+    } else {
+        bldr_url_from_env_load_or_default()
+    }
+}
+
+pub fn bldr_auth_token_from_args_env_or_load(opt: Option<String>) -> Result<String, Error> {
+    if let Some(token) = opt {
+        Ok(token)
+    } else {
+        match henv::var(AUTH_TOKEN_ENVVAR) {
+            Ok(v) => Ok(v),
+            Err(_) => {
+                match config::load()?.auth_token {
+                    Some(v) => Ok(v),
+                    None => {
+                        Err(Error::ArgumentError("No auth token specified. Please \
+                                                  check that you have specified a \
+                                                  valid Personal Access Token with:  \
+                                                  -z, --auth <AUTH_TOKEN>"
+                                                                          .into()))
+                    }
+                }
+            }
+        }
+    }
+}
+
+pub fn maybe_bldr_auth_token_from_args_or_load(opt: Option<String>) -> Option<String> {
+    match bldr_auth_token_from_args_env_or_load(opt) {
+        Ok(v) => Some(v),
+        Err(_) => None,
+    }
 }
 
 lazy_static! {
     pub static ref CACHE_KEY_PATH_DEFAULT: String =
         hab_core_fs::CACHE_KEY_PATH.to_string_lossy().to_string();
+    pub static ref HAB_BLDR_URL_DEFAULT: String = bldr_url_from_env_load_or_default();
```

Comment:
It does not look like this is currently being used? Could we add it back in a subsequent PR when we know precisly how we are going to handle it?